core/src/main/java/org/apache/cxf/common/spi/ClassLoaderProxyService.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.cxf.common.spi;
+
+import java.util.Map;
+
+import org.apache.cxf.Bus;
+
+public class ClassLoaderProxyService implements ClassLoaderService {
+    NamespaceClassCreator srv;
+    public ClassLoaderProxyService(Bus bus) {
+        this(new NamespaceClassGenerator(bus));
+    }
+    public ClassLoaderProxyService(NamespaceClassCreator srv) {
+        this.srv = srv;
+    }
+
+    @Override
+    public Object createNamespaceWrapper(Class<?> mcls, Map<String, String> map) {
+        Class<?> cls = srv.createNamespaceWrapper(mcls, map);
+        try {
+            return cls.getConstructor(Map.class).newInstance(map);
+        } catch (Throwable e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+    public class LoadFirst extends ClassLoaderProxyService {
+        public LoadFirst(Bus bus) {
+            //TODO not sure here if I get class loader like that ???
```

Comment:
Should we try to get the `Classloader` from `bus`?

```
ClassLoader classloader = bus.getExtension(ClassLoader.class); 
```

@rmannibucau wouldn't fallback to ` Thread.currentThread().getContextClassLoader();` be more appropriate here?

Reply:
It hink we have a classloaderutils which tries the tccl and if null uses the system classloader. Would be a saner default if the bus does not have one.

Reply:
@reta maybe we can skip bus param and class loader parameter and just use in GeneratedNamespaceClassLoader
ClassLoaderUtils.loadClass(String className, Class<?> callingClass) with callingClass equal to GeneratedNamespaceClassLoader.class
and we use same pattern for all generated class. wdyt ?
@rmannibucau  @reta

Reply:
@dufoli any classloader usage should following this chain 1. bus one, 2. tccl, 3. system classloader (there is an utility for 2+3). Using any random class classloader will lead to a random behavior in all but flat classpath env sadly.

Reply:
hmm I have use ClassLoaderUtils . It mean that all classloader service provider must start with bus instead of only ClassLoaderUtils.load ? It mean I have to get set bus in contructor and check bus.getExtension(Classloader) right ?

Reply:
May be we could introduce overloaded method to `ClassLoaderUtils` which also accepts `bus` and does the classloader resolution taking into account the extension?

Reply:
I have just push a solution with a common class to do that with bus

Reply:
👍 , no objections, thank you